slackbridge/utils.py

```diff
@@ -36,6 +74,54 @@ def strip_nick(nick):
     ])
 
 
+def format_irc_message(text, users, channels):
+    def chan_replace(match):
+        chan_id = match.group(1)
+        readable = match.group(2)
+        return '#{}'.format(readable or channels[chan_id])
+
+    def user_replace(match):
+        user_id = match.group(1)
+        readable = match.group(2)
+        return readable or users[user_id].nickname
+
+    def cmd_replace(match):
+        command = match.group(1)
+        label = match.group(2)
+        return '<{}>'.format(label or command)
+
+    def emoji_replace(match):
+        emoji = match.group(1)
+        if emoji in EMOJIS:
+            return EMOJIS[emoji]
+        return ':{}:'.format(emoji)
+
+    text = re.sub(r'\n|\r', ' ', text)
+    text = text.replace('<!channel>', '@channel') \
+               .replace('<!everyone>', '@everyone') \
+               .replace('<!here>', '@here')
+
+    # Replace channels, users, commands, links, emoji, and any remaining stuff
+    # Adapted from
+    # https://github.com/ekmartin/slack-irc/blob/2b5ceb7ca7beb/lib/bot.js#L154
+    text = re.sub(r'<#(C\w+)\|?(\w+)?>', chan_replace, text)
+    text = re.sub(r'<\@(U\w+)\|?(\w+)?>', user_replace, text)
+    text = re.sub(r'<!(\w+)\|?(\w+)?>', cmd_replace, text)
+    text = re.sub(r'<(?!!)([^|]+?)>', lambda match: match.group(1), text)
+    text = re.sub(r':([\d\w+-]+):', emoji_replace, text)
+    text = re.sub(r'<.+?\|(.+?)>', lambda match: match.group(1), text)
+
+    text = text.replace('&lt;', '<').replace('&gt;', '>').replace('&amp;', '&')
+    return text
+
+
+def format_slack_message(text):
+    # Strip any color codes coming from IRC, since Slack cannot display them
+    # Current solution is taken from https://stackoverflow.com/a/970723
+    # TODO: Preserve bold and italics (convert to markdown?)
+    return re.sub(r'\x03(?:\d{1,2}(?:,\d{1,2})?)?', '', text, flags=re.UNICODE)
+
```

Comment:
I think Slack also requires that &, <, and > be [HTML-encoded](https://api.slack.com/docs/message-formatting).

Reply:
Their docs do say that, but I believe the client library we are using to interface with Slack already does this.

Reply:
Ah, cool.

Reply:
I can't find exactly where it does, but we haven't had any issues with sending these characters to Slack, and I tested them yesterday too.

Reply:
Ok, I've added a comment about this, since I think it's a good thing to mention.